### PR TITLE
fix(web): correctly compute localized downtime's datepickers

### PR DESCRIPTION
**Pull request by sc979**
## Description

Fix an issue where datetime of downtimes weren't correctly computed, when not using the english localization :
- the ending datepicker was not set properly and displayed unconsistent date
- adding a downtime on two days, did not actualize the starting date of the datepicker
- setting an end time before the start date was possible and resulted on useless downtime (not processed by centengine)

**Fixes** MON-3935 & https://github.com/centreon/centreon/pull/7606 from @UrBnW  + https://github.com/centreon/centreon/issues/5963 + https://github.com/centreon/centreon/issues/7337 + https://github.com/centreon/centreon/issues/8266 + #8362 + #8383.

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x
- [ ] 18.10.x
- [x] 19.04.x
- [x] 19.10.x
- [x] 20.04.x (master)

<h2> How this pull request can be tested ? </h2>

- Choose for example the french language.
- Select a default duration on the Administration > parameters > monitoring + duration
- Set a downtime from the monitoring > status > downtime + add downtime
    check that the start and end date and time are set properly.
    check that consistent values are displayed when changing the datepicker and timepicker values.
    check that the datepicker and timepicker are still consistent with the choosen language.

- Do the same from all the other available downtime form in the UI eg :
    status > monitoring > services + choose a resource + set a downtime from the toolbar
    status > monitoring > host + choose a resource + set a downtime from the toolbar
    ...

- Check that no errors are displayed in the logs

### CODE REVIEW

File: SECURITY.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
+# Security Policy
```
Comment (sc979) [inline SECURITY.md]: seems that this file was already present on my branch. But is not related to this PR

File: host-monitoring/src/action.php [hunk 2]
```diff
@@ -239,70 +224,72 @@
...
+                    ? localStorage.getItem('realTimezone')
+                    : moment.tz.guess()).format("HH:mm")
+                );
+            }
+        });
+        jQuery("#start_time, #end_time").timepicker();
+        initDatepicker();
+        turnOnEvents();
+        updateEndTime();
+    });
```
Comment (sc979) [inline host-monitoring/src/action.php]: only this part was modified in this file. all the other modifications are related to coding style

File: host-monitoring/src/downtime.ihtml [hunk 3]
```diff
@@ -21,18 +19,21 @@
...
 		<tr>
 			<td class="FormRowField">{$durationLabel}</td>
 			<td class="FormRowValue">
-				<input type='text' name='dayduration' size='2' value='{$defaultDayDuration}' />&nbsp;{$daysLabel}
-				<input type='text' name='hourduration' size='2' value='{$defaultHourDuration}' />&nbsp;{$hoursLabel}
-				<input type='text' name='minuteduration' size='2' value='{$defaultMinuteDuration}' />&nbsp;{$minutesLabel}
-                <input type='text' name='secondduration' size='5' value='{$defaultSecondDuration}' />&nbsp;{$secondsLabel}
+				<input type='text' id='duration' name='duration' width='30' value='{$defaultDuration}' />
+				<select id="duration_scale" name="duration_scale">
+					<option value="s" {$sDefaultScale}>Seconds</option>
```
Comment (kduret) [inline host-monitoring/src/downtime.ihtml]: translation seems not managed anymore

File: host-monitoring/src/action.php [hunk 4]
```diff
@@ -72,20 +74,19 @@
...
-    $duration = $defaultDuration;
-    if ($defaultScale == 'm') {
-        $duration *= 60;
-    } elseif ($defaultScale == 'h') {
-        $duration *= 3600;
-    } elseif ($defaultScale == 'd') {
-        $duration *= 86400;
+    if (
+        isset($centreon->optGen['monitoring_dwt_duration'])
+        && $centreon->optGen['monitoring_dwt_duration']
```
Comment (callapa) [inline host-monitoring/src/action.php]: What is the purpose of this test?

Comment (callapa) [reply host-monitoring/src/action.php]: Maybe you can replace the both conditions by !empty(...)

File: host-monitoring/src/action.php [hunk 5]
```diff
@@ -72,20 +74,19 @@
...
+    ) {
+        $defaultDuration = $centreon->optGen['monitoring_dwt_duration'];
+        if (
+            isset($centreon->optGen['monitoring_dwt_duration_scale'])
+            && $centreon->optGen['monitoring_dwt_duration_scale']
```
Comment (callapa) [inline host-monitoring/src/action.php]: Same comments as above

File: host-monitoring/src/action.php [hunk 6]
```diff
@@ -138,39 +140,22 @@
             $template->assign('process_service_checked', $process_service_checked);
 
             $force_active_checked = '';
-            if (isset($centreon->optGen['monitoring_ack_active_checks']) &&
-                $centreon->optGen['monitoring_ack_active_checks']
+            if (
+                isset($centreon->optGen['monitoring_ack_active_checks'])
+                && $centreon->optGen['monitoring_ack_active_checks']
```
Comment (callapa) [inline host-monitoring/src/action.php]: Same comments as above

File: host-monitoring/src/action.php [hunk 7]
```diff
@@ -112,8 +113,9 @@
 
             /* default ack options */
             $persistent_checked = '';
-            if (isset($centreon->optGen['monitoring_ack_persistent']) &&
-                $centreon->optGen['monitoring_ack_persistent']
+            if (
+                isset($centreon->optGen['monitoring_ack_persistent'])
+                && $centreon->optGen['monitoring_ack_persistent']
```
Comment (callapa) [inline host-monitoring/src/action.php]: Same comments as above

File: host-monitoring/src/action.php [hunk 8]
```diff
@@ -239,70 +224,72 @@
...
-}
+    function sendCmd()
+    {
+        fieldResult = true;
+        if (jQuery("#comment") && !jQuery("#comment").val()) {
+            fieldResult = false;
+        }
+        if (fieldResult == false) {
+            jQuery("#result").html("<font color=red><b>Please fill all mandatory fields.</b></font>");
+            return false;
```
Comment (callapa) [inline host-monitoring/src/action.php]: Why return a boolean value ? This value will not be used anywhere.

File: host-monitoring/src/action.php (hunk 4 above)
Comment (UrBnW) [reply host-monitoring/src/action.php]: Here we simply verify that value is set (first test) and different than 0 (second test).

File: host-monitoring/src/action.php (hunk 8 above)
Comment (UrBnW) [reply host-monitoring/src/action.php]: _Not that it's just a style change here, not related to this PR's goal..._

File: host-monitoring/src/downtime.ihtml (hunk 3 above)
Comment (UrBnW) [reply host-monitoring/src/downtime.ihtml]: My bad yes, sorry for this...
I added required fix to the [related _services_ PR](https://github.com/centreon/centreon-widget-service-monitoring/pull/129) :
https://github.com/centreon/centreon-widget-service-monitoring/pull/129/commits/f27df1773bfe6631a0027aa0d698f9ab19884a50

File: host-monitoring/src/action.php (hunk 8 above)
Comment (sc979) [reply host-monitoring/src/action.php]: indeed. My IDE corrected the indentation of the file. @callapa , should I revert it ?

**Pull request merged**